nef-models/Command+URLQueryItem.swift

```diff
@@ -0,0 +1,41 @@
+//  Copyright © 2020 The nef Authors.
+
+import Foundation
+
+public extension Command {
+    var item: URLQueryItem {
+        switch self {
+        case .exportSnippetToFile(let selection):
+            return URLQueryItem(name: key, value: selection)
+        case .exportSnippetToClipboard(let selection):
+            return URLQueryItem(name: key, value: selection)
+        case .markdownPage(let playground):
+            return URLQueryItem(name: key, value: playground)
+        case .playgroundBook(let package):
+            return URLQueryItem(name: key, value: package)
+        case .preferences, .about, .notification:
+            return URLQueryItem(name: key, value: "")
+        }
+    }
+}
+
+public extension URLQueryItem {
+    var command: Command? {
+        switch name {
+        case Command.about.key:
+            return .about
+        case Command.preferences.key:
+            return .preferences
+        case Command.exportSnippetToFile(selection: "").key:
+            return value.flatMap(Command.exportSnippetToFile)
+        case Command.exportSnippetToClipboard(selection: "").key:
+            return value.flatMap(Command.exportSnippetToClipboard)
+        case Command.markdownPage(playground: "").key:
+            return value.flatMap(Command.markdownPage)
+        case Command.playgroundBook(package: "").key:
+            return value.flatMap(Command.playgroundBook)
+        default:
+            return nil
+        }
```

Comment:
This does not look very good to me. You have to create a value of a `Command`, in some cases providing no data, just to obtain a key.

Reply:
Well, it was to use the same model as in extension menu as in the command together with selection, code... but we can break it into two models, one to read extension menu action and the other to send the action, they are two different models but related 1:1. Anyway, I agree with you it is weird and smells they are conceptually two models.